File: Leetcode/1184.js
/**
 * 1184. 公交站间的距离
 * link：https://leetcode.cn/problems/distance-between-bus-stops/
 */

/**
 * @param {number[]} distance
 * @param {number} start
 * @param {number} destination
 * @return {number}
 */
var distanceBetweenBusStops = function (distance, start, destination) {
    let leftDistance = 0, rightDistance = 0;
    const n = distance.length;
    for (let i = start; i != destination; i = (i + 1) % n) {
        leftDistance += distance[i];
    }
    for (let i = (start + n - 1) % n; i != (destination + n - 1) % n; i = (i + n - 1) % n) {
        rightDistance += distance[i];
    }
    return Math.min(leftDistance, rightDistance);
};

File: Leetcode/1184.cpp
/**
 * 1184. 公交站间的距离 Distance between bus stops
 * link：https://leetcode.cn/problems/distance-between-bus-stops
 */

#include <algorithm>
#include <vector>

using namespace std;

class Solution {
public:
    int distanceBetweenBusStops(vector<int>& distance, int start, int destination) {
        if (start == destination) {
            return 0;
        }
        int n = distance.size();
        int result = 0;
        int sum = distance[start];
        for (int i = (start + 1) % n; i != start; i = (i + 1) % n) {
            if (i == destination) {
                result = sum;
            }
            sum += distance[i];
        }
        return min(result, sum - result);
    }
};